src/boot/efi/linux.c

```diff
@@ -15,9 +15,61 @@
 #include "linux.h"
 #include "missing_efi.h"
 #include "pe.h"
+#include "secure-boot.h"
 #include "util.h"
 
-EFI_STATUS load_trusted_image(EFI_HANDLE parent, const void *source, size_t len, EFI_HANDLE *ret_image) {
+static struct TrustedState {
+        EFI_SECURITY_FILE_AUTHENTICATION_STATE orig_auth;
+        EFI_SECURITY2_FILE_AUTHENTICATION orig_auth2;
```

Comment:
btw, no big fan of needless abbreviation in structures. Just call this original_authentication_protocol and original_authentication2_protocol or so?